### PR TITLE
Remove side effect of ActiveQuery::getTablesUsedInFrom() introduced in 2.0.13

### PR DESCRIPTION
**Pull request by terales**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #15353

### CODE REVIEW

**Comment by samdark** [comment]
Tests aren't passing.

**Comment by terales** [comment]
@samdark thanks for attention,

This was done by intention to help @cebe easier check what's going on with this comment: https://github.com/yiisoft/yii2/issues/15353#issuecomment-351456253

I'm waiting for clarification from @cebe. If this side-effect was introduced accidentally than I would make tests pass, but if there is some deep idea than I would close PR and remove branch.

**Comment by samdark** [comment]
Ah... let me take a look. @cebe is very busy now.

**Comment by samdark** [comment]
Oh, yes. This side effect wasn't intended, I think.

**Comment by terales** [comment]
Sure!

**Comment by terales** [comment]
Checked Codeclimat failures: they are connected to the extracting table names clearing in another method. I think refactoring of this function should be in another issue if you feel that it should be refactored.

**Comment by samdark** [comment]
It is fine as is.

File: framework/db/Query.php [hunk 1]
```diff
@@ -474,7 +474,17 @@ public function getTablesUsedInFrom()
...
 
-        // Clean up table names and aliases
+        return $this->cleanUpTableNames($tableNames);
+    }
+
+    /**
+     * Clean up table names and aliases
+     * Both aliases and names are enclosed into {{ and }}.
+     * @param array $tableNames non-empty array
+     * @return string[] table names indexed by aliases
```
Comment (samdark) [inline framework/db/Query.php]: Need since tag.

**Review by samdark — APPROVED** [review]
All good. Need changelog and one since tag in phpdoc. Otherwise good to merge.

**Comment by terales** [comment]
Done!

Thanks a lot for your hard work on Yii in general and with pushing modular approach into the core in 2.1!

**Comment by samdark** [comment]
Merged. Thank you!

**Pull request merged**